common/src/main/java/io/netty/util/ResourceLeakDetector.java

```diff
@@ -245,18 +245,24 @@ public final ResourceLeak open(T obj) {
         return track0(obj);
     }
 
-    @SuppressWarnings("unchecked")
-    private DefaultResourceLeak track0(T obj) {
+    protected boolean shouldTrack(T obj) {
         Level level = ResourceLeakDetector.level;
         if (level == Level.DISABLED) {
-            return null;
+            return false;
         }
 
         if (level.ordinal() < Level.PARANOID.ordinal()) {
             if ((PlatformDependent.threadLocalRandom().nextInt(samplingInterval)) == 0) {
```

Comment:
This should just be a return right?

`return (PlatformDependent.threadLocalRandom().nextInt(samplingInterval)) == 0);`